ggdeploymentd/src/component_config.c

```diff
@@ -0,0 +1,198 @@
+// aws-greengrass-lite - AWS IoT Greengrass runtime for constrained devices
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "component_config.h"
+#include "deployment_model.h"
+#include <assert.h>
+#include <ggl/buffer.h>
+#include <ggl/core_bus/gg_config.h>
+#include <ggl/error.h>
+#include <ggl/list.h>
+#include <ggl/log.h>
+#include <ggl/map.h>
+#include <ggl/object.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+static GglError apply_reset_config(
+    GglBuffer component_name, GglMap component_config_map
+) {
+    GglObject *reset_configuration = NULL;
+    GglError ret = ggl_map_validate(
+        component_config_map,
+        GGL_MAP_SCHEMA(
+            { GGL_STR("reset"), false, GGL_TYPE_LIST, &reset_configuration }
+        )
+    );
+    if (ret != GGL_ERR_OK) {
+        return ret;
+    }
+
+    // If there is no reset configuration, then there is no
+    // configuration update to make
+    if (reset_configuration == NULL) {
+        return GGL_ERR_OK;
+    }
+
+    if (reset_configuration->type != GGL_TYPE_LIST) {
+        GGL_LOGE("Reset update did not parse into a list "
+                 "during configuration updates.");
+        return GGL_ERR_INVALID;
+    }
+    GGL_LIST_FOREACH(reset_element, reset_configuration->list) {
```

Comment:
what do these `reset_element`s look like?

If I want to reset a key c with path a/b/c , wouldn't we need to parse that path into a list like [ a , b , c ] to pass to the config delete like [ services , component , configuration , a , b , c ] ?

It seems like since we're only using it as a buffer you can only reset non-nested keys?